src/commissaire_service/investigator/__init__.py

```diff
@@ -177,16 +177,24 @@ def on_investigate(self, message, address, cluster_data={}):
             # Save the updated host model.
             self.storage.save(host)
 
-        # Verify association with relevant container managers
-        # FIXME Adapt this for ContainerManagerConfig.
-        # params = {
-        #     'cluster_type': cluster.type,
-        #     'address': address
-        # }
-        # response = self.request('storage.node_registered', params=params)
-        # if response['result']:
-        #     host.status = 'active'
-        host.status = 'disassociated'  # XXX Temporary hack.
+        # Register with container manager (if applicable).
+        try:
+            host.status = 'disassociated'
+            if cluster.container_manager:
+                response = self.request(
+                    'container.register_node',
+                    cluster.container_manager, address)
+                if response['result']:
```

Comment:
FYI ``register_node`` won't return. It will raise on error ([ref](https://github.com/projectatomic/commissaire/blob/master/src/commissaire/containermgr/kubernetes/__init__.py#L224-L231))